---
title: C28105
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: f6a0b92c67840343a7ae1ea0986b27c76375e2e1
ms.sourcegitcommit: 174c992ecdc868ecbf7d3cee654bbc2855aeb67d
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 12/06/2019
ms.locfileid: "74879070"
---
# <a name="c28105"></a>C28105
警告 C28105：因例外狀況而洩漏資源

 當引發例外狀況時，不會釋放指定的資源。 路徑所指定的語句可能會引發例外狀況。 此警告類似于警告[C28103](../code-quality/c28103.md)，但在此情況下，會牽涉到例外狀況。

## <a name="example"></a>範例
 下列程式碼範例會產生此警告：

```cpp
res = KeSaveFloatingPointState(buffer);

res = AllocateResource(Resource);
char *p2 = new char[10]; // could throw

delete[] p2;
FreeResource(Resource)
```

 下列程式碼範例可避免此警告：

```cpp
res = AllocateResource(Resource);
char *p2;

try {
    p2 = new char[10];
} catch (std::bad_alloc *e) {
    // just handle the throw
    ;
}
FreeResource(Resource)
```
